ecs-cli/modules/cli/regcreds/regcreds_app.go

```
@@ -31,7 +46,184 @@ func Up(c *cli.Context) {
+			hasSecretARN = true
+		}
+
+		log.Infof("Processing credentials for registry %s...", registryName)
+
+		if hasCredPair && hasSecretARN {
+			arn, err := updateOrWarnForExistingSecret(credentialEntry, updateAllowed, smClient)
+			if err != nil {
+				return nil, err
+			}
+			registryResults[registryName] = buildOutputEntry(arn, credentialEntry.ContainerNames)
+
+		} else if hasSecretARN {
+			registryResults[registryName] = buildOutputEntry(credentialEntry.SecretManagerARN, credentialEntry.ContainerNames)
+			log.Infof("Using existing secret %s.", credentialEntry.SecretManagerARN)
+
+		} else {
+			arn, err := createNewRegistrySecret(registryName, credentialEntry, smClient)
+			if err != nil {
+				return nil, err
+			}
+			registryResults[registryName] = buildOutputEntry(arn, credentialEntry.ContainerNames)
+		}
+	}
+
+	return &registryResults, nil
+}
+
+func createNewRegistrySecret(registryName string, credEntry readers.RegistryCredEntry, smClient secretsClient.SMClient) (string, error) {
+
+	secretName := generateSecretName(registryName)
+
+	existingSecret, _ := smClient.DescribeSecret(*secretName)
+	if existingSecret != nil {
+		log.Infof("Existing credential secret found, using %s", *existingSecret.ARN)
+
+		return *existingSecret.ARN, nil
+	}
+
+	secretString := generateSecretString(credEntry.Username, credEntry.Password)
+
+	createSecretRequest := secretsmanager.CreateSecretInput{
+		Name:         secretName,
+		SecretString: secretString,
+		Description:  generateSecretDescription(registryName),
+	}
+	if credEntry.KmsKeyID != "" {
+		createSecretRequest.SetKmsKeyId(credEntry.KmsKeyID)
+	}
+
+	output, err := smClient.CreateSecret(createSecretRequest)
+	if err != nil {
+		return "", err
+	}
+	log.Infof("New credential secret created: %s", *output.ARN)
+
+	return *output.ARN, nil
+}
+
+func updateOrWarnForExistingSecret(credEntry readers.RegistryCredEntry, updateAllowed bool, smClient secretsClient.SMClient) (string, error) {
```

Since this doesn't appear to actually change the `secretArn`, could we just have this method return an error instead of having a multiple value return?